include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,690 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_fe_interface_values_h
+#define dealii_fe_interface_values_h
+
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping_q1.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * FEInterfaceValues is a data structure to access and assemble Finite Element
+ * data on interfaces between two cells of a mesh.
+ *
+ * It provides a way to access average and jump terms used in
+ * Discontinuous Galerkin methods on faces between two neighboring cells. In
+ * the literature these are called "inner interfaces" or "facets".
+ *
+ * Internally, this class provides an abstraction for two FEFaceValues
+ * objects. The class introduces a new "interface dof index" that walks over
+ * the union of the dof indices of the two FEFaceValues objects. Helper
+ * functions allow translating between the new "interface dof index" and the
+ * corresponding "fe index" and "dof index" of the FE.
+ *
+ * The class is made to be used inside MeshWorker::mesh_loop. It is intended
+ * to be a low level replacement for MeshWorker and LocalIntegrators and a
+ * higher level abstraction compared to assembling face terms manually.
+ *
+ *
+ * @author Timo Heister, 2019.
+ */
+template <int dim, int spacedim = dim>
+class FEInterfaceValues
+{
+public:
+  /**
+   * Construct an FEInterfaceValues from existing FEFaceValues objects.
+   */
+  FEInterfaceValues(const FEFaceValues<dim> &   fe,
+                    const FESubfaceValues<dim> &fe_sub,
+                    const FEFaceValues<dim> &   fe_neighbor,
+                    const FESubfaceValues<dim> &fe_sub_neighbor);
```

Comment:
Shouldn't there also be a constructor that just takes two FEValues objects? In fact, it's not clear to me why you need *four* objects here...

Reply:
I think you are correct that there is no reason to supply different FE*Values for the neighbor here as the 4 internal objects are copied from the arguments anyways.

Reply:
One could use this class to compute interface terms with different FEs on both sides as required for hp, but I have not tested this. Anyways, I changed the constructor.